### DISCUSSION TITLE
A couple of interesting questions about Commons

### DISCUSSION THREAD

**Discussion by Artem13327** [FossifyOrg/General-Discussion#88] [General]
If a person wants to build the old version, then what should he do with this library if he needs to get exactly the same version?
Will it be cleaned from being associated with the pro version?
Does this library have any version designation?

**Comment by inson1** [comment]
I have another question. Where is the logic about changing logo color? thanks

**Reply by naveensingh** [reply]
@inson1 what logo? Do you mean the Commons Sample app icon?

**Comment by naveensingh** [comment]
> Does this library have any version designation?

No. 

> If a person wants to build the old version, then what should he do with this library if he needs to get exactly the same version?

Assuming you are talking about the library itself, you can just use `git reset --hard desired_commit_hash` and then build the library at any point in the past. You can go to https://jitpack.io/#org.fossify/commons/ if you want to find the latest build. Same story if you want to use an old version of the commons library in any of the Fossify apps.

> Will it be cleaned from being associated with the pro version?

If you are talking about code and logic related to pro versions, all pro-related things such as 'Upgrade to pro' popups, etc. have been disabled. I haven't removed it yet because if donations fail to sustain the project, we'll need a few pro versions.

**Comment by Artem13327** [comment]
> You can go to https://jitpack.io/#org.fossify/commons/ if you want to find the latest build. Same story if you want to use an old version of the commons library in any of the Fossify apps.

As I noticed, there is at least some division into versions, so why not give the user at least some idea about this, so that if necessary, he can repeat the assembly of any version

**Reply by naveensingh** [reply]
> As I noticed, there is at least some division into versions

Those aren't exactly versions. Every time someone changes something in the library, a new commit hash is generated. The random strings you see on that website are abbreviated versions of the commit hashes. There is no strict preference for when to update the library and when to avoid it.

> why not give the user at least some idea about this, so that if necessary, he can repeat the assembly of any version

To "repeat the assembly" of any version of any Fossify app, usually, all you need to do is just reset the code to the desired version and build the app. The older commons library referenced by the older version of the code will be downloaded from Jitpack. E.g. if you want to build [version 1.0.0](https://github.com/FossifyOrg/Gallery/releases/tag/1.0.0) of the Gallery app, all you need to do is run `git reset --hard 1.0.0` and then build the app.

**Reply by Artem13327** [reply]
I found where the version of this library is indicated, I just didn’t see the version indicated in some source code, the source of which I don’t remember (Maybe she was there too)
